### DISCUSSION TITLE
Similar tools

### DISCUSSION THREAD

**Discussion by j-brooke** [j-brooke/FracturedJson#6] [General]
If FracturedJson doesn't do quite what you need, or isn't available in your preferred environment, this is the place to talk about alternatives.

**Comment by j-brooke** [comment]
[OjG](https://github.com/ohler55/ojg) is a JSON library for the Go language that, among many other features, includes pretty-printing with smart inlining of complex elements.

**Reply by ohler55** [reply]
FracturedJson does a nice job of formatting output for the areas it covers.

**Comment by j-brooke** [comment]
[NeatJSON](http://phrogz.net/js/neatjson/neatjson.html) provides a variety of formatting options, including smart inlining based on length.  It's available for Ruby, Lua, and JavaScript.

**Comment by jon-freed** [comment]
* https://github.com/lydell/json-stringify-pretty-compact
* jq with --compact-output, see https://jqlang.org/manual/#invoking-jq
* https://github.com/deftek/json_pp

(I'm not recommending any of these; just cross-referencing)

**Comment by j-brooke** [comment]
[Prettier](https://prettier.io/) is probably the best known formatting library anywhere.  With JSON, here's what I've observed.  (Please correct me if I'm wrong - I may have missed some settings.)

* If the source document is already expand expanded, it won't compress it.  It likes to leave any line breaks where they are.
* If the source is condensed (minified, for example) it will inline objects/arrays to fit in the defined line length.  (I don't think it considers complexity.)

**Comment by smmilton** [comment]
I'm not a typescript guy, but loved the idea you had with FracturedJson in general. Since my preferred language is Perl, here is a Perl re-implementation of the FracturedJson 5.0 library, along with a perl CLI tool to use it.

https://github.com/smmilton/FracturedJsonPerl

**Comment by j-brooke** [comment]
[jsonfold](https://github.com/yairlenga/jsonfold) shares the same core goal as FracturedJson: producing compact yet highly readable JSON. It uses line-length and nesting limits to fold small structures, supports a table/grid layout for similar rows, and emphasizes a streaming architecture (a notable difference from FracturedJson’s document-tree approach). Multiple languages are supported and development is moving quickly.

**Comment by j-brooke** [comment]
[json-stringify-pretty-compact](https://github.com/lydell/json-stringify-pretty-compact) (with [CLI](https://github.com/avantgardnerio/json-stringify-pretty-compact-cli) and [online demo](https://lydell.github.io/json-stringify-pretty-compact/)) and its Python port [jsonreflow](https://github.com/soxofaan/jsonreflow) take a lighter-touch approach: they keep short arrays and objects on a single line (respecting a max length) while falling back to standard pretty-printing for larger structures. Simple and effective for many use cases.
